src/main/java/com/felixgrund/codeshovel/parser/AbstractFunction.java

```diff
@@ -77,7 +77,7 @@ private String getIdParameterString() {
 	}
 
 	protected String getInitialId(E rawMethod) {
-		String ident = getName();
+		String ident = getParentName() + "#" + getName();
```

Comment:
The problem here was that a nested method had the same function ID as another method with the same name and parameters. In the end, there was only one function mapped to the ID and hence the wrong function was taken as "matched function". By adding the parent name to the function ID, nested functions will differ from top-level functions.